## 62% of end-end tests have passed
```
Total executed tests: 8
Failed tests: 2
Expected to fail tests: 1
Passed tests: 5
```
```
IDs of failed tests: 702745,703133 
```
```
IDs of expected to fail tests: 702843 
```
<h3>Failed tests (2)</h3><details><summary>Click to expand</summary><br/><li><a href="https://ci.status.im/job/status-mobile/job/e2e/job/status-app-prs-rerun/parambuild/?BRANCH_NAME=seanstrom/password-max-length&PR_ID=21593&APK_NAME=21593.apk&TR_CASE_IDS=702745,703133">Rerun failed tests</a></li><br/><h4>Class TestOneToOneChatMultipleSharedDevicesNewUi:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3714200">test_1_1_chat_non_latin_messages_stack_update_profile_photo</a>, id: 702745 </b></td></tr><tr><td>

```
Test setup failed: critical/chats/test_1_1_public_chats.py:20: in prepare_devices
    self.drivers, self.loop = create_shared_drivers(2)
base_test_case.py:330: in create_shared_drivers
    raise e
base_test_case.py:320: in create_shared_drivers
    test_suite_data.current_test.testruns[-1].jobs[drivers[i].session_id] = i + 1
 '_asyncio.Future' object has no attribute 'session_id'
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/68d004de1b3249e69c616ec801461b0a?auth=9d5307381f9c09857302cc7e6184187f">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3714202">test_restore_multiaccount_with_waku_backup_remove_profile_switch</a>, id: 703133 </b></td></tr><tr><td><p><blockquote><div>Device 1: Find `Button` by `accessibility id`: `show-profiles` </div><div>Device 1: Tap on found: Button </div></blockquote></p>

```
critical/chats/test_public_chat_browsing.py:247: in test_restore_multiaccount_with_waku_backup_remove_profile_switch
    self.errors.verify_no_errors()
base_test_case.py:192: in verify_no_errors
    pytest.fail('\n '.join([self.errors.pop(0) for _ in range(len(self.errors))]))
 zQ3...dWXh5 was not restored as a contact from waku backup!
E    zQ3...Vacac was not restored as a contact from waku backup!
E    admin_open was not restored from waku-backup!!
E    member_open was not restored from waku-backup!!
E    admin_closed was not restored from waku-backup!!
E    member_closed was not restored from waku-backup!!
```

<br/><br/><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/afe723db2c424ecd9ca97e77d7c33e57?auth=ea35fc882a19f8fc4255c0b0b9c86430#88">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details><h3>Expected to fail tests (1)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestCommunityMultipleDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3714201">test_community_message_edit</a>, id: 702843 </b></td></tr><tr><td>

```
Test is not run, e2e blocker  
```

<b>[[reason: [NOTRUN] Skipped due to waku issue on staging fleet]]</b><br/><br/></td></tr></tbody></table></details><h3>Passed tests (5)</h3><details><summary>Click to expand</summary><br/><br/><h4>Class TestWalletMultipleDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3714204">test_wallet_send_asset_from_drawer</a>, id: 727230 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3714203">test_wallet_send_eth</a>, id: 727229 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestWalletOneDevice:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3714205">test_wallet_add_remove_regular_account</a>, id: 727231 </b></td></tr><tr><td></td></tr><tr><td><b>2. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3714206">test_wallet_balance_mainnet</a>, id: 740490 </b></td></tr><tr><td></td></tr></tbody></table><h4>Class TestCommunityOneDeviceMerged:</h4><table style="width: 100%"><colgroup><col span="1" style="width: 20%;"><col span="1" style="width: 80%;"></colgroup><tbody><tr></tr><tr><td><b>1. <a href="https://ethstatus.testrail.net/index.php?/tests/view/3714199">test_community_copy_and_paste_message_in_chat_input</a>, id: 702742 </b></td></tr><tr><td><ins>Device sessions</ins><p><ul><p>Device 1:<ul><li><a href="https://eu-central-1.saucelabs.com/jobs/afe723db2c424ecd9ca97e77d7c33e57?auth=ea35fc882a19f8fc4255c0b0b9c86430#56">Steps, video, logs</a></li></ul></p></ul></p></td></tr></tbody></table></details>

--- a/src/status_im/common/password_with_hint/style.cljs
+++ b/src/status_im/common/password_with_hint/style.cljs
@@ -1,0 +1,4 @@
+(ns status-im.common.password-with-hint.style)
+
+(def info-message
+  {:margin-top 8})

--- a/src/status_im/common/password_with_hint/view.cljs
+++ b/src/status_im/common/password_with_hint/view.cljs
@@ -1,0 +1,25 @@
+(ns status-im.common.password-with-hint.view
+  (:require
+    [quo.core :as quo]
+    [quo.foundations.colors :as colors]
+    [react-native.core :as rn]
+    [status-im.common.password-with-hint.style :as style]))
+
+(defn view
+  [{{:keys [text status shown?]} :hint :as input-props}]
+  (tap> shown?)
+  [rn/view
+   [quo/input
+    (-> input-props
+        (dissoc :hint)
+        (assoc :type  :password
+               :blur? true))]
+   [rn/view {:style style/info-message}
+    (when shown?
+      [quo/info-message
+       (cond-> {:status status
+                :size   :default}
+         (not= :success status) (assoc :icon :i/info)
+         (= :success status)    (assoc :icon :i/check-circle)
+         (= :default status)    (assoc :color colors/white-70-blur))
+       text])]])

--- a/src/status_im/common/password_with_hint/view.cljs
+++ b/src/status_im/common/password_with_hint/view.cljs
@@ -7,8 +7,7 @@
 
 (defn view
   [{{:keys [text status shown?]} :hint :as input-props}]
-  (tap> shown?)
-  [rn/view
+  [:<>
    [quo/input
     (-> input-props
         (dissoc :hint)

--- a/src/status_im/common/validation/password.cljs
+++ b/src/status_im/common/validation/password.cljs
@@ -5,14 +5,13 @@
 
 (defn validate-short-enough?
   [password]
-  (utils.string/at-least-n-chars? password
-                                  constants/new-password-min-length))
+  (utils.string/at-most-n-chars? password
+                                 constants/new-password-max-length))
 
 (defn validate-long-enough?
   [password]
-  (and (validate-short-enough? password)
-       (utils.string/at-most-n-chars? password
-                                      constants/new-password-max-length)))
+  (utils.string/at-least-n-chars? password
+                                  constants/new-password-min-length))
 
 (defn validate
   [password]
@@ -21,8 +20,17 @@
                      utils.string/has-upper-case?
                      utils.string/has-numbers?
                      utils.string/has-symbols?
-                     validate-short-enough?
-                     validate-long-enough?)]
+                     validate-long-enough?
+                     validate-short-enough?)]
     (->> password
          validations
-         (zipmap constants/password-tips))))
+         (zipmap (conj constants/password-tips
+                       :long-enough?
+                       :short-enough?)))))
+
+(defn strength
+  [validations]
+  (->> (select-keys validations constants/password-tips)
+       (vals)
+       (filter true?)
+       count))

--- a/src/status_im/constants.cljs
+++ b/src/status_im/constants.cljs
@@ -133,6 +133,7 @@
 (def ^:const min-password-length 6)
 (def ^:const pincode-length 6)
 (def ^:const new-password-min-length 10)
+(def ^:const new-password-max-length 100)
 (def ^:const max-group-chat-participants 20)
 (def ^:const max-group-chat-name-length 24)
 (def ^:const default-number-of-messages 20)

--- a/src/status_im/constants.cljs
+++ b/src/status_im/constants.cljs
@@ -143,9 +143,7 @@
   [:lower-case?
    :upper-case?
    :numbers?
-   :symbols?
-   :long-enough?
-   :short-enough?])
+   :symbols?])
 
 (def ^:const strength-status
   {1 :very-weak

--- a/src/status_im/contexts/onboarding/create_password/view.cljs
+++ b/src/status_im/contexts/onboarding/create_password/view.cljs
@@ -6,6 +6,7 @@
     [react-native.platform :as platform]
     [react-native.safe-area :as safe-area]
     [status-im.common.floating-button-page.view :as floating-button]
+    [status-im.common.password-with-hint.view :as password-with-hint]
     [status-im.constants :as constants]
     [status-im.contexts.onboarding.create-password.style :as style]
     [utils.i18n :as i18n]
@@ -27,27 +28,9 @@
      :size   :paragraph-1}
     (i18n/label :t/password-creation-subtitle)]])
 
-(defn password-with-hint
-  [{{:keys [text status shown]} :hint :as input-props}]
-  [rn/view
-   [quo/input
-    (-> input-props
-        (dissoc :hint)
-        (assoc :type  :password
-               :blur? true))]
-   [rn/view {:style style/info-message}
-    (when shown
-      [quo/info-message
-       {:status status
-        :size   :default
-        :icon   (if (= status :success) :i/check-circle :i/info)
-        :color  (when (= status :default)
-                  colors/white-70-blur)}
-       text])]])
-
 (defn password-inputs
   [{:keys [passwords-match? on-change-password on-change-repeat-password on-input-focus
-           password-long-enough? empty-password? show-password-validation?
+           password-long-enough? password-short-enough? empty-password? show-password-validation?
            on-blur-repeat-password]}]
   (let [hint-1-status (if password-long-enough? :success :default)
         hint-2-status (if passwords-match? :success :error)
@@ -58,19 +41,24 @@
                            (not passwords-match?)
                            (not empty-password?))]
     [:<>
-     [password-with-hint
-      {:hint           {:text   (i18n/label :t/password-creation-hint)
-                        :status hint-1-status
-                        :shown  true}
+     [password-with-hint/view
+      {:hint           (if (not password-short-enough?)
+                         {:text   (i18n/label
+                                   :t/password-creation-max-length-hint)
+                          :status :error
+                          :shown? true}
+                         {:text   (i18n/label :t/password-creation-hint)
+                          :status hint-1-status
+                          :shown? true})
        :placeholder    (i18n/label :t/password-creation-placeholder-1)
        :on-change-text on-change-password
        :on-focus       on-input-focus
        :auto-focus     true}]
      [rn/view {:style style/space-between-inputs}]
-     [password-with-hint
+     [password-with-hint/view
       {:hint           {:text   hint-2-text
                         :status hint-2-status
-                        :shown  (and (not empty-password?)
+                        :shown? (and (not empty-password?)
                                      show-password-validation?)}
        :error?         error?
        :placeholder    (i18n/label :t/password-creation-placeholder-2)
@@ -100,10 +88,11 @@
                           utils.string/has-upper-case?
                           utils.string/has-numbers?
                           utils.string/has-symbols?
-                          #(utils.string/at-least-n-chars? % constants/new-password-min-length))]
+                          #(utils.string/at-least-n-chars? % constants/new-password-min-length)
+                          #(utils.string/at-most-n-chars? % constants/new-password-max-length))]
     (->> password
          validations
-         (zipmap (conj constants/password-tips :long-enough?)))))
+         (zipmap (conj constants/password-tips :long-enough? :short-enough?)))))
 
 (defn calc-password-strength
   [validations]
@@ -115,12 +104,13 @@
   [password]
   (rn/use-memo
    (fn []
-     (let [{:keys [long-enough?]
+     (let [{:keys [long-enough? short-enough?]
             :as   validations} (validate-password password)]
-       {:password-long-enough? long-enough?
-        :password-validations  validations
-        :password-strength     (calc-password-strength validations)
-        :empty-password?       (empty? password)}))
+       {:password-long-enough?  long-enough?
+        :password-short-enough? short-enough?
+        :password-validations   validations
+        :password-strength      (calc-password-strength validations)
+        :empty-password?        (empty? password)}))
    [password]))
 
 (defn- use-repeat-password-checks
@@ -174,20 +164,18 @@
 
 
         {:keys [password-long-enough?
+                password-short-enough?
                 password-validations password-strength
                 empty-password?]}                       (use-password-checks password)
 
         {:keys [same-password-length? same-passwords?]} (use-repeat-password-checks password
                                                                                     repeat-password)
 
-        meet-requirements?                              (rn/use-memo
-                                                         #(and (not empty-password?)
-                                                               (utils.string/at-least-n-chars? password
-                                                                                               10)
-                                                               same-passwords?
-                                                               accepts-disclaimer?)
-                                                         [password repeat-password
-                                                          accepts-disclaimer?])]
+        meet-requirements?                              (and (not empty-password?)
+                                                             password-long-enough?
+                                                             password-short-enough?
+                                                             same-passwords?
+                                                             accepts-disclaimer?)]
 
     [floating-button/view
      {:header [page-nav]
@@ -231,6 +219,7 @@
       [header]
       [password-inputs
        {:password-long-enough?     password-long-enough?
+        :password-short-enough?    password-short-enough?
         :passwords-match?          same-passwords?
         :empty-password?           empty-password?
         :show-password-validation? show-password-validation?

--- a/src/status_im/contexts/onboarding/create_password/view.cljs
+++ b/src/status_im/contexts/onboarding/create_password/view.cljs
@@ -82,12 +82,6 @@
     [quo/tips {:completed? symbols?}
      (i18n/label :t/password-creation-tips-4)]]])
 
-(defn calc-password-strength
-  [validations]
-  (->> (vals validations)
-       (filter true?)
-       count))
-
 (defn- use-password-checks
   [password]
   (rn/use-memo
@@ -97,7 +91,7 @@
        {:password-long-enough?  long-enough?
         :password-short-enough? short-enough?
         :password-validations   validations
-        :password-strength      (calc-password-strength validations)
+        :password-strength      (password/strength validations)
         :empty-password?        (empty? password)}))
    [password]))
 

--- a/src/status_im/contexts/profile/settings/screens/password/change_password/new_password_form.cljs
+++ b/src/status_im/contexts/profile/settings/screens/password/change_password/new_password_form.cljs
@@ -2,8 +2,8 @@
   (:require
     [clojure.string :as string]
     [quo.core :as quo]
-    [quo.foundations.colors :as colors]
     [react-native.core :as rn]
+    [status-im.common.password-with-hint.view :as password-with-hint]
     [status-im.constants :as constant]
     [status-im.contexts.profile.settings.screens.password.change-password.events]
     [status-im.contexts.profile.settings.screens.password.change-password.header :as header]
@@ -12,24 +12,6 @@
     [utils.re-frame :as rf]
     [utils.security.core :as security]
     [utils.string :as utils.string]))
-
-(defn- password-with-hint
-  [{{:keys [text status shown?]} :hint :as input-props}]
-  [:<>
-   [quo/input
-    (-> input-props
-        (dissoc :hint)
-        (assoc :type  :password
-               :blur? true))]
-   [rn/view {:style style/info-message}
-    (when shown?
-      [quo/info-message
-       (cond-> {:status status
-                :size   :default}
-         (not= :success status) (assoc :icon :i/info)
-         (= :success status)    (assoc :icon :i/check-circle)
-         (= :default status)    (assoc :color colors/white-70-blur))
-       text])]])
 
 (defn- calc-password-strength
   [validations]
@@ -76,11 +58,15 @@
         long-enough?                                   (utils.string/at-least-n-chars?
                                                         password
                                                         constant/new-password-min-length)
+        short-enough?                                  (utils.string/at-most-n-chars?
+                                                        password
+                                                        constant/new-password-max-length)
         empty-password?                                (string/blank? password)
         same-passwords?                                (and (not empty-password?)
                                                             (= password repeat-password))
         meet-requirements?                             (and (not empty-password?)
                                                             long-enough?
+                                                            short-enough?
                                                             same-passwords?
                                                             disclaimer-accepted?)
         error?                                         (and show-validation?
@@ -115,17 +101,21 @@
     [:<>
      [rn/scroll-view {:style style/form-container}
       [header/view]
-      [password-with-hint
-       {:hint           {:text   (i18n/label :t/password-creation-hint)
-                         :status (if long-enough? :success :default)
-                         :shown? true}
+      [password-with-hint/view
+       {:hint           (if (not short-enough?)
+                          {:text   (i18n/label :t/password-creation-max-length-hint)
+                           :status :error
+                           :shown? true}
+                          {:text   (i18n/label :t/password-creation-hint)
+                           :status (if long-enough? :success :default)
+                           :shown? true})
         :placeholder    (i18n/label :t/change-password-new-password-placeholder)
         :label          (i18n/label :t/change-password-new-password-label)
         :on-change-text on-change-password
         :on-focus       on-input-focus
         :auto-focus     true}]
       [rn/view {:style style/space-between-inputs}]
-      [password-with-hint
+      [password-with-hint/view
        {:hint           {:text   (if same-passwords?
                                    (i18n/label :t/password-creation-match)
                                    (i18n/label :t/password-creation-dont-match))

--- a/src/status_im/contexts/profile/settings/screens/password/change_password/new_password_form.cljs
+++ b/src/status_im/contexts/profile/settings/screens/password/change_password/new_password_form.cljs
@@ -13,16 +13,10 @@
     [utils.re-frame :as rf]
     [utils.security.core :as security]))
 
-(defn- calc-password-strength
-  [validations]
-  (->> (vals validations)
-       (filter true?)
-       count))
-
 (defn- help
   [{:keys [validations]}]
   (let [{:keys [lower-case? upper-case? numbers? symbols?]} validations
-        password-strength                                   (calc-password-strength validations)]
+        password-strength                                   (password/strength validations)]
     [rn/view
      [quo/strength-divider {:type (constant/strength-status password-strength :info)}
       (i18n/label :t/password-creation-tips-title)]

--- a/src/utils/string.cljs
+++ b/src/utils/string.cljs
@@ -42,6 +42,10 @@
   [s n]
   (>= (count s) n))
 
+(defn at-most-n-chars?
+  [s n]
+  (<= (count s) n))
+
 (defn safe-trim
   [s]
   (when (string? s)

--- a/translations/en.json
+++ b/translations/en.json
@@ -1900,6 +1900,7 @@
   "password-creation-dont-match": "Passwords do not match",
   "password-creation-hint": "Minimum 10 characters",
   "password-creation-match": "Passwords match",
+  "password-creation-max-length-hint": "Maximum 100 characters",
   "password-creation-placeholder-1": "Type password",
   "password-creation-placeholder-2": "Repeat password",
   "password-creation-subtitle": "To log in to Status and sign transactions",

